### PR TITLE
Typescript 3

### PR DESCRIPTION
**Pull request by annehaley**
Continuing the typescript conversion, this is the next chunk of converted files.

### CODE REVIEW

File: client/src/components/NavigationTabs.vue [hunk 1]
```diff
@@ -10,20 +10,20 @@ export default defineComponent({
...
+    const store = useStore();
     const user = inject('user') as User;
+    const currentDatasetId = computed(
+      () => store.state.currentDatasetId,
+    );
+
     return {
       user,
+      currentDatasetId,
+      setDrawer: () => store.commit('setDrawer'),
```
Comment (dchiquito) [inline client/src/components/NavigationTabs.vue:22]:
```suggestion
      setDrawer: () => store.commit.setDrawer,
```

File: client/src/components/DataImportExport.vue [hunk 2]
```diff
@@ -1,46 +1,53 @@
...
-    async importData() {
-      this.importing = true;
-      this.importErrorText = '';
-      this.importErrors = false;
+  setup() {
+    const store = useStore();
+
+    const mainSession = inject('mainSession') as Session;
+    const loadSession = (session: Session) => store.dispatch('loadSession', session);
+    const loadLocalDataset = (files: FileList) => store.dispatch('loadLocalDataset', files);
```
Comment (dchiquito) [inline client/src/components/DataImportExport.vue:15]:
```suggestion
    const loadSession = (session: Session) => store.dispatch.loadSession(session);
    const loadLocalDataset = (files: FileList) => store.dispatch.loadLocalDataset(files);
```

File: client/src/django.ts [hunk 3]
```diff
@@ -56,7 +57,7 @@ const djangoClient = new Vue({
       const { data } = await apiClient.get(`/sessions/${sessionId}/settings`);
       return data;
     },
-    async setSettings(sessionId: string, settings) {
+    async setSettings(sessionId: string, settings: Object) {
```
Comment (dchiquito) [inline client/src/django.ts:60]: Something more descriptive would be good. I think it's just `{importpath: string, exportpath: string}`. Could you add that to `types.ts`?

**Review by dchiquito — CHANGES_REQUESTED** [review]

File: client/src/components/JSONConfig.vue [hunk 4]
```diff
@@ -1,45 +1,61 @@
...
-      if (!this.$refs.form.validate()) {
+  setup() {
+    const mainSession = inject('mainSession') as Session;
+
+    const importpath = ref('');
+    const exportpath = ref('');
+    const changed = ref(false);
+    const importpathError = ref('');
+    const exportpathError = ref('');
+    const form = ref(null);
```
[the remark this answers is not in the document]
Comment (annehaley) [reply client/src/components/JSONConfig.vue:16]: No errors are raised for it without the explicit `as`. Should we add that in anyway?

File: client/src/components/EmailDialog.vue [hunk 5]
```diff
@@ -20,6 +22,25 @@ export default {
       default: () => [],
     },
   },
+  setup() {
+    const store = useStore();
+    const screenshots = computed(() => store.state.screenshots);
```
[the remark this answers is not in the document]
Comment (annehaley) [reply client/src/components/EmailDialog.vue]: it does work, and the linter offered that object decomposition is preferred, so I changed lines like this to
`const { screenshots } = store.state;`

File: client/src/components/JSONConfig.vue (hunk 4 above)
[the remark this answers is not in the document]
Comment (dchiquito) [reply client/src/components/JSONConfig.vue:16]: The type cast is something of a last resort, so if everything works without it then everything is fine

**Review by dchiquito — APPROVED** [review]

**Pull request merged**